src/main/scala/firrtl/passes/MemUtils.scala

```diff
@@ -64,71 +63,67 @@ object toBits {
 // TODO: make easier to understand
 object toBitMask {
   def apply(e: Expression, dataType: Type): Expression = e match {
-    case ex: WRef => hiermask(ex, ex.tpe, dataType)
-    case ex: WSubField => hiermask(ex, ex.tpe, dataType)
-    case ex: WSubIndex => hiermask(ex, ex.tpe, dataType)
+    case ex @ (_: WRef | _: WSubField | _: WSubIndex) => hiermask(ex, ex.tpe, dataType)
     case t => error("Invalid operand expression for toBits!")
   }
-  def hiermask(e: Expression, maskType: Type, dataType: Type): Expression = (maskType, dataType) match {
-    case (mt: VectorType, dt: VectorType) => seqCat((0 until mt.size).reverse.map(i => hiermask(WSubIndex(e, i, mt.tpe, UNKNOWNGENDER), mt.tpe, dt.tpe)))
-    case (mt: BundleType, dt: BundleType) => seqCat((mt.fields zip dt.fields).map { case (mf, df) =>
-      hiermask(WSubField(e, mf.name, mf.tpe, UNKNOWNGENDER), mf.tpe, df.tpe) } )
+  private def hiermask(e: Expression, maskType: Type, dataType: Type): Expression = (maskType, dataType) match {
+    case (mt: VectorType, dt: VectorType) => seqCat(0 until mt.size map { i =>
+      hiermask(WSubIndex(e, i, mt.tpe, UNKNOWNGENDER), mt.tpe, dt.tpe) })
+    case (mt: BundleType, dt: BundleType) => seqCat(mt.fields zip dt.fields map { case (mf, df) =>
+      hiermask(WSubField(e, mf.name, mf.tpe, UNKNOWNGENDER), mf.tpe, df.tpe) })
```

Comment:
I would much rather these take more lines  for clarity: eg.
```scala
    case (mt: VectorType, dt: VectorType) => 
      seqCat((0 until mt.size) map { i =>
        hiermask(WSubIndex(e, i, mt.tpe, UNKNOWNGENDER), mt.tpe, dt.tpe)
      })
    case (mt: BundleType, dt: BundleType) =>
      seqCat((mt.fields zip dt.fields) map { case (mf, df) =>
        hiermask(WSubField(e, mf.name, mf.tpe, UNKNOWNGENDER), mf.tpe, df.tpe)
      })
```
(I also tend to prefer parentheses around range creation and zipping)

Also, in the original code there was a reverse on the range, why was it removed? is this correct without it?